lcls-twincat-motion/Library/POUs/Motion/FB_MotionStage.TcPOU

```diff
@@ -69,8 +72,16 @@ IF stMotionStage.bError THEN
 	stMotionStage.bEnable := FALSE;
 END_IF
 
-// Mark done if settled, then reset the FB if there is no error
-IF stMotionStage.Axis.Status.InTargetPosition AND fbDriveVirtual.bDone THEN
+// Check the limits and cancel execution if appropriate. Without this block we have infinite error spam
+bFwdHit := stMotionStage.Axis.Status.PositiveDirection AND NOT stMotionStage.bLimAllFwd;
```

Comment:
You're using this combination to account for any unforseen halt in a given direction?

Reply:
Yes. If any kind of forward limit is triggered while moving forward, we note `bFwdHit` and engage the brake appropriately.

Reply:
The crucial point is that `Status.PositiveDirection` has to be there to ensure we can move off of the limit 😉